### DISCUSSION TITLE
Realtime performance optimization

### DISCUSSION THREAD

**Discussion by jtayl711** [apohl79/audiogridder#1158] [Q&A]
My main use case for Audiogridder is for real-time performance.  What can I do to optimize performance and minimize latency?  Audio quality is less important to me than round-trip latency.  I currently have the server on an i7 PC and both the plugin and server are connected to a gigabit switch w/ cat5e cabling.  Any other suggestions?

**Comment by emack999** [comment]
I also use AG for real-time performance.  My suggestion to you is to use it wisely, on effects such as delays, reverbs etc. that can afford a bit of latency without affecting playability.
Using AG for real-time instrument plugins can work, depending on the instrument and sound.  Keyboard pad and string sounds are fine with it---e-drums are borderline too much latency for real-time performance.  Guitar also.

**Reply by jtayl711** [reply]
Great advice, thank you.

**Comment by jaacMusic** [comment]
My best suggestion for your AG server is to do the following:

- Disable all the Window services you don't need (Google a service functionality before doing this!!)
- Remove/disable startup applications.
- Under "Advanced System Settings" > "Performance" settings, select adjust for best performance.
- Also under "Advance System Settings", disable "System Protection"
- Update your network drivers or better yet, get server level network card.
- Don't use WiFi for network connectivity
- Don't have Antivirus software installed and disable Windows Defender
- If you server has plenty of RAM, consider disabling the system page file (make everything run 100% from RAM)
- Unplug unnecessary USB devices from AG server (they take CPU clock cycles and can introduce audio drops/latency).
- Be smart about the DSP plugins you want to use (some are designed for live/zero latency scenarios. Use those instead)
- You will get more CPU if you disable the Sandbox mode in AG server.
- Get or use a good audio interface with low latency (All audio interfaces will have some latency by nature. I use DigiGrid D which gives me a 0. 8ms. Most other audio interfaces will have as high as 2-3ms; it adds up!).

Over all, make sure your AG server is just for AG processing if you want to best results. There are various videos in YouTube on to tweak your Windows PC for latency and audio production.
Hope all this helps.

**Reply by jtayl711** [reply]
This is great advice, thank you.  I'm already doing many of these things.  I'm not too familiar with disabling unneeded windows services.  Do you recommend using an LTSC version of Windows?

**Reply by jaacMusic** [reply]
Sure! LTSC would be a good approach. Just make sure to check your plugin dependencies before deciding to go with a Windows LTSC version.

**Reply by MythicalCheese** [reply]
(What I take to be) the creator wrote that having Sandboxing On doesn't make AG perform any worse:
https://github.com/apohl79/audiogridder/discussions/614

But you're saying it'll save CPU. If I weren't assuming the perosn in that thread made AG (or maintains it), then I'd assume you're correct about the CPU. Thoughts? 

I'm always looking to make my system more efficient, but am very new to AG. I don't understand some of these settings.
